search/tasks.py

```diff
@@ -209,6 +209,63 @@ def index_courses(ids):
         return error
 
 
+@app.task(autoretry_for=(RetryException,), retry_backoff=True, rate_limit="600/m")
+def index_bootcamps(ids):
+    """
+    Index bootcamps
+
+    Args:
+        ids(list of int): List of bootcamp id's
+
+    """
+    try:
+        api.index_bootcamps(ids)
+    except (RetryException, Ignore):
+        raise
+    except:  # pylint: disable=bare-except
+        error = f"index_bootcamps threw an error"
+        log.exception(error)
+        return error
+
+
+@app.task(autoretry_for=(RetryException,), retry_backoff=True, rate_limit="600/m")
+def index_programs(ids):
+    """
+    Index programs
+
+    Args:
+        ids(list of int): List of program id's
+
+    """
+    try:
+        api.index_programs(ids)
+    except (RetryException, Ignore):
+        raise
+    except:  # pylint: disable=bare-except
+        error = f"index_programs threw an error"
+        log.exception(error)
+        return error
+
+
+@app.task(autoretry_for=(RetryException,), retry_backoff=True, rate_limit="600/m")
+def index_learning_paths(ids):
```

Comment:
`index_*` functions like `index_bootcamps`, `index_programs`, `index_learning_paths` , etc can be refactored into a shared function. Should we do so, or keep it separate as `index_{profile,comment,post,course}` are currently?

Reply:
It might be easier to keep separate, since they all used different serializers/models

Reply:
:+1: